api/v1alpha3/types.go

```diff
@@ -321,6 +323,42 @@ type UserAssignedIdentity struct {
 	ProviderID string `json:"providerID"`
 }
 
+const (
+	// AzureIdentityBindingSelector is the label used to match with the AzureIdentityBinding
+	// For the controller to match an identity binding, it needs a [label] with the key `aadpodidbinding`
+	// whose value is that of the `selector:` field in the `AzureIdentityBinding`.
+	AzureIdentityBindingSelector = "capz-controller-aadpodidentity-selector"
+)
+
+// IdentityType represents different types of identities.
+type IdentityType string
+
+const (
+	// UserAssignedMSI represents a user-assigned identity.
+	UserAssignedMSI IdentityType = "UserAssignedMSI"
+
+	// ServicePrincipal represents a service principal.
+	ServicePrincipal IdentityType = "ServicePrincipal"
+)
+
+// IdentitySpec specifies the parameters that are used to create an AzureIdentity
+type IdentitySpec struct {
```

Comment:
We had discussed having `AllowedNamespaces` label selector on the `IdentitySpec`, similar to the design approach in CAPA: https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/docs/proposal/20200506-single-controller-multitenancy.md#namespace-restrictions

This could be added in a subsequent PR.

Reply:
I started adding it, will submit it with the rest of the comments